### PR TITLE
events-processor-add-eventSource-apigateway

### PR DESCRIPTION
**Pull request by kaerukoara**
*Description of changes:*

1. add  API Gateway Event Sources 
2. add Events Notification data  via apigateway invoke by curl or by IoT Core & Https rule

*Evidence:*
SnapCrab_NoName_2021-9-10_23-48-24_No-00 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*********************************************************************************************************************
```
Comment (boardthatpowder) [inline source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.service.ts:1]: To be consistent, as this controller is dealing with messages, move it to source/packages/services/events-processor/src/api/messages/apigwtrigger.service.ts.

**Review by boardthatpowder — CHANGES_REQUESTED** [review]
In addition - please create change logs (run `rush change` from `source` and enter details of changes for the prompted projects).

File: source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { interfaces, controller, response, requestBody, httpPost} from 'inversify-express-utils';
+import { inject } from 'inversify';
+import {TYPES} from '../../di/types';
+import {logger} from '../../utils/logger.util';
+import {handleError} from '../../utils/errors.util';
+import { ApigwTriggerService } from './apigwtrigger.service';
+import { CommonEvent } from '../../transformers/transformers.model';
+
+@controller('')
+export class ApigwTriggerController implements interfaces.Controller {
```
[the remark this answers is not in the document]
Comment (kaerukoara) [reply source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts]: Thanks! I fixed it according to your suggestion.

File: source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    constructor( @inject(TYPES.ApigwTriggerService) private apigwtriggerService: ApigwTriggerService) {}
+
+    @httpPost('/apigwtrigger')
```
[the remark this answers is not in the document]
Comment (kaerukoara) [reply source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts]: Thanks! I fixed it according to your suggestion.

File: source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    public async apigwTrigger(@requestBody() event:CommonEvent, @response() res: Response) : Promise<void> {
```
[the remark this answers is not in the document]
Comment (kaerukoara) [reply source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.controller.ts]: Thanks! I fixed it according to your suggestion.

File: source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.service.ts (hunk 1 above)
Comment (kaerukoara) [reply source/packages/services/events-processor/src/api/apigwtrigger/apigwtrigger.service.ts]: Thanks! I fixed it according to your suggestion.

**Comment by kaerukoara** [comment]
> In addition - please create change logs (run `rush change` from `source` and enter details of changes for the prompted projects).
 I added it according to your suggestion. Thank you.
And I have confirmed the new api path is ok by the following test.
Screen Shot 2021-09-13 at 14 01 15

**Review by boardthatpowder — APPROVED** [review]

**Pull request merged**